Lib/test/test_configparser.py

```diff
@@ -691,7 +700,21 @@ def test_write(self):
                     "option-without-value\n"
                     "\n"
                     )
-            self.assertEqual(output.getvalue(), expect_string)
+            if testfile is None:
+                observed = output.getvalue()
+            else:
+                with open(testfile, 'r') as testf:
+                    observed = testf.read()
+            self.assertEqual(observed, expect_string)
+
+    def test_write_filename(self):
+        f = tempfile.NamedTemporaryFile(delete=False)
+        try:
+            self.test_write(testfile=f.name)
+        except:
+            raise
+        finally:
+            os.unlink(f.name)
```

Comment:
`support.unlink()` has better support under Windows so it handles issues like the one Terry posted on the python-dev thread better. However, I wonder if there's a better option than `NamedTemporaryFile` (I just woke up so my brain doesn't work yet)

Reply:
We can avoid try...finally block if we use [`self.addCleanup()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup).